src/acpi-tables/src/madt.rs

```diff
@@ -28,13 +29,23 @@ pub struct LocalAPIC {
 }
 
 impl LocalAPIC {
-    pub fn new(cpu_id: u8) -> Self {
-        Self {
-            r#type: 0,
-            length: 8,
-            processor_uid: cpu_id,
-            apic_id: cpu_id,
-            flags: U32::new(1u32 << MADT_CPU_ENABLE_FLAG),
+    pub fn new(cpu_id: u8, online_capable: bool) -> Self {
+        if online_capable {
+            Self {
+                r#type: 0,
+                length: 8,
+                processor_uid: cpu_id,
+                apic_id: cpu_id,
+                flags: U32::new(1u32 << MADT_CPU_ONLINE_CAPABLE_FLAG),
+            }
+        } else {
+            Self {
+                r#type: 0,
+                length: 8,
+                processor_uid: cpu_id,
+                apic_id: cpu_id,
+                flags: U32::new(1u32 << MADT_CPU_ENABLE_FLAG),
+            }
```

Comment:
I think it's more natural to say whether a CPU is enabled or not and make everything else online capable. Also, you can write this more concisely:

```Rust
pub fn new(cpu_id: u8, online: bool) -> Self {
    let flags = if online {
          U32::new(1u32 << MADT_CPU_ENABLE_FLAG)
    } else {
          U32::new(1u32 << MADT_CPU_ONLINE_CAPABLE_FLAG)
    };
    Self {
        r#type: 0,
        length: 8,
        processor_uid: cpu_id,
        apic_id: cpu_id,
        flags
    }
}
```